addons/mass_mailing/controllers/main.py

```diff
@@ -115,7 +115,7 @@ def blacklist_check(self, mailing_id, res_id, email, token):
         if not self._valid_unsubscribe_token(mailing_id, res_id, email, token):
             return 'unauthorized'
         if email:
-            record = request.env['mail.blacklist'].sudo().with_context(active_test=False).search([('email', '=ilike', email)])
+            record = request.env['mail.blacklist'].sudo().with_context(active_test=False).search([('email', '=', email)])
```

Comment:
As in previous commit you should not take email as a valid input. You should normalize it before handling it.